### PR TITLE
Use "Cargo feature" language everywhere and other tutorial tweaks

### LINKED ISSUE #2710: Resolve sffc post-submit tutorial feedback
I left comments on https://github.com/unicode-org/icu4x/pull/2678 and https://github.com/unicode-org/icu4x/pull/2674.

### LINKED ISSUE #2754: Datagen docs should be clearer about experimental keys
We should improve our datagen docs to make it clearer what experimental is and what it does.

Currently it leads to confusions like https://github.com/unicode-org/icu4x/issues/2664

### PR DESCRIPTION
**Pull request by sffc**
_I don't expect reviews on PRs I send this week until January_

Fixes #2710
Fixes #2754
Part of #2853

### CODE REVIEW

File: docs/tutorials/data_management.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ fn main() {
 
 With this provider, we can use the `unstable` constructors. These are only guaranteed to work if the data was generated with the same version of ICU4X that you are building with, but if you build the data as part of your a build pipeline, that shouldn't be a problem.
 
-You can also implement the `AnyProvider` trait, so that it can be used with `_with_any_provider` constructors. Using these constructors is slightly less performant than the `unstable` ones, but, as the name suggests, stable across (minor) releases.
+You can also implement the `AnyProvider` trait, so that it can be used with `_with_any_provider` constructors. Using these constructors is slightly less performant than the `unstable` ones, and it doesn't perform automatic data slicing, but, as the name suggests, it is stable across (minor) releases.
```
Comment (robertbastian) [inline docs/tutorials/data_management.md:187]: the unstable constructors don't "perform" data slicing either, but they "support" it

File: docs/tutorials/intro.md [hunk 2]
```diff
@@ -114,7 +112,7 @@ Users are also free to design their own providers that best fit into their ecosy
 
 # 5. Using an ICU4X component
 
-We're going to extend our app to use the `icu::datetime` component to format a date and time. This component requires data, but as we don't want to jump into data management just yet, we will use `ICU4X`'s `icu_testdata` crate. This contains test providers that support all ICU4X keys for a small representative set of locales. It contains both a `BufferProvider` (`icu_testdata::buffer()`), and an `AnyProvider` (`icu_testdata::any()`). The latter requires fewer features, so we will be using that.
+We're going to extend our app to use the `icu::datetime` component to format a date and time. This component requires data, but as we don't want to jump into data management just yet, we will use `ICU4X`'s `icu_testdata` crate. This contains test providers that support all ICU4X keys for a small representative set of locales. It contains a `BufferProvider` (`icu_testdata::buffer()`), an `AnyProvider` (`icu_testdata::any()`), and a `BakedProvider` (`icu_testdata::unstable()`). The latter two require fewer features, so we will be using those.
```
Comment (robertbastian) [inline docs/tutorials/intro.md:115]: `BakedProvider` is not a type. It's called `UnstableDataProvider` in `icu_testdata`, but it's `#[doc(hidden)]`.

```suggestion
We're going to extend our app to use the `icu::datetime` component to format a date and time. This component requires data, but as we don't want to jump into data management just yet, we will use `ICU4X`'s `icu_testdata` crate. This contains test providers that support all ICU4X keys for a small representative set of locales. It contains a `BufferProvider` (`icu_testdata::buffer()`), an `AnyProvider` (`icu_testdata::any()`), and an unstable data provider (`icu_testdata::unstable()`). The latter two require fewer features, so we will be using those.
```

File: docs/tutorials/intro.md [hunk 3]
```diff
@@ -132,17 +130,17 @@ use icu::datetime::{DateTimeFormatter, options::length};
 const LOCALE: Locale = locale!("ja"); // let's try some other language
 
 fn main() {
-    let provider = icu_testdata::any();
+    let provider = icu_testdata::unstable();
```
Comment (robertbastian) [inline docs/tutorials/intro.md:133]: I think this is better with `AnyProvider`. `unstable` sounds scary, and explaining what it means, how it's more efficient, how it supports data slicing, is beyond the scope of the intro. Any is "nice".

File: docs/tutorials/data_management.md (hunk 1 above)
Comment (sffc) [reply docs/tutorials/data_management.md:187]: "allow for" seems right

File: docs/tutorials/intro.md (hunk 2 above)
Comment (sffc) [reply docs/tutorials/intro.md]: ok

File: docs/tutorials/intro.md (hunk 3 above)
Comment (sffc) [reply docs/tutorials/intro.md:133]: I wish we could agree on a better term than `unstable` because it really shouldn't be that scary. But ok.

**Review by Manishearth — APPROVED** [review]
i like this!

**Comment by sffc** [comment]
Merging this due to risk of bitrot, but if @robertbastian has any feedback I can address it in another PR

File: docs/tutorials/intro.md (hunk 3 above)
Comment (robertbastian) [reply docs/tutorials/intro.md]: Maybe `versioned` would have been a better name?

**Comment by robertbastian** [comment]
lgtm

**Pull request merged**